src/test/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminAreasOfLawServiceTest.java

```diff
@@ -44,4 +46,120 @@ void shouldReturnAllAreasOfLaw() {
 
         assertThat(areasOfLawService.getAllAreasOfLaw()).isEqualTo(expectedResult);
     }
+
+    @Test
+    void shouldReturnAnAreaOfLawForGivenId() {
+        final AreaOfLaw mockAreaOfLaw = getTestAreaOfLawEntities().get(0);
+        when(areasOfLawRepository.getOne(100)).thenReturn(mockAreaOfLaw);
+
+        final uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw expectedResult =
+            new uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw(mockAreaOfLaw);
+
+        assertThat(areasOfLawService.getAreaOfLaw(100)).isEqualTo(expectedResult);
+    }
+
+    @Test
+    void getAreaOfLawShouldThrowNotFoundExceptionIfIdDoesNotExist() {
+        when(areasOfLawRepository.getOne(400)).thenThrow(javax.persistence.EntityNotFoundException.class);
+        assertThatThrownBy(() -> areasOfLawService
+            .getAreaOfLaw(400))
+            .isInstanceOf(NotFoundException.class);
+    }
+
+    @Test
+    void shouldUpdateAreaOfLaw() {
+        final uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw areaOfLaw =
+            new uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw(getTestAreaOfLawEntities().get(0));
+        areaOfLaw.setExternalLink("https://something.else.link");
+        areaOfLaw.setExternalLinkDescription("A different description");
+        areaOfLaw.setExternalLinkDescriptionCy("A different description in Welsh");
+
+        when(areasOfLawRepository.findById(areaOfLaw.getId()))
+            .thenReturn(Optional.of(getTestAreaOfLawEntities().get(0)));
+        when(areasOfLawRepository.save(any(AreaOfLaw.class)))
+            .thenAnswer((Answer<AreaOfLaw>) invocation -> invocation.getArgument(0));
```

Comment:
I wonder what is the benefit or using 'thenAnswer' here compared to using the simple 'thenReturn' since it will return a fixed value from save and we are mocking the return of a fixed value. Would the below not work?

        final AreaOfLaw entity = getTestAreaOfLawEntities().get(0);
        when(areasOfLawRepository.findById(areaOfLaw.getId())).thenReturn(Optional.of(entity));
        when(areasOfLawRepository.save(entity)).thenReturn(entity);
        assertThat(areasOfLawService.updateAreaOfLaw(areaOfLaw)).isEqualTo(areaOfLaw);